neo.UnitTests/Ledger/UT_MemoryPool.cs

```diff
@@ -253,5 +287,205 @@ public void TestInvalidateAll()
             _unit.UnverifiedSortedTxCount.ShouldBeEquivalentTo(30);
             _unit.SortedTxCount.ShouldBeEquivalentTo(0);
         }
+
+        [TestMethod]
+        public void TestContainsKey()
+        {
+            AddTransactions(10);
+
+            var txToAdd = CreateTransaction();
+            _unit.TryAdd(txToAdd.Hash, txToAdd);
+            _unit.ContainsKey(txToAdd.Hash).Should().BeTrue();
+            _unit.InvalidateVerifiedTransactions();
+            _unit.ContainsKey(txToAdd.Hash).Should().BeTrue();
+        }
+
+        [TestMethod]
+        public void TestGetEnumerator()
+        {
+            AddTransactions(10);
+            _unit.InvalidateVerifiedTransactions();
+            IEnumerator<Transaction> enumerator = _unit.GetEnumerator();
+            foreach (Transaction tx in _unit)
+            {
+                enumerator.MoveNext();
+                enumerator.Current.Should().BeSameAs(tx);
+            }
+        }
+
+        [TestMethod]
+        public void TestIEnumerableGetEnumerator()
+        {
+            AddTransactions(10);
+            _unit.InvalidateVerifiedTransactions();
+            IEnumerable enumerable = _unit;
+            var enumerator = enumerable.GetEnumerator();
+            foreach (Transaction tx in _unit)
+            {
+                enumerator.MoveNext();
+                enumerator.Current.Should().BeSameAs(tx);
+            }
+        }
+
+        [TestMethod]
+        public void TestGetVerifiedTransactions()
+        {
+            var tx1 = CreateTransaction();
+            var tx2 = CreateTransaction();
+            _unit.TryAdd(tx1.Hash, tx1);
+            _unit.InvalidateVerifiedTransactions();
+            _unit.TryAdd(tx2.Hash, tx2);
+            IEnumerable<Transaction> enumerable = _unit.GetVerifiedTransactions();
+            enumerable.Count().Should().Be(1);
+            var enumerator = enumerable.GetEnumerator();
+            enumerator.MoveNext();
+            enumerator.Current.Should().BeSameAs(tx2);
+        }
+
+        [TestMethod]
+        public void TestReVerifyTopUnverifiedTransactionsIfNeeded()
+        {
+            NeoSystem TheNeoSystem = TestBlockchain.InitializeMockNeoSystem();
+            var s = Blockchain.Singleton.Height;
+            _unit = new MemoryPool(TheNeoSystem, 600);
+            _unit.LoadPolicy(TestBlockchain.GetStore().GetSnapshot());
+            AddTransactions(2);
+            _unit.InvalidateVerifiedTransactions();
+            AddTransactions(513);
+            var result = _unit.ReVerifyTopUnverifiedTransactionsIfNeeded(10, Blockchain.Singleton.GetSnapshot());
+            _unit.VerifiedCount.Should().Be(514);
+            _unit.UnVerifiedCount.Should().Be(1);
+            result.Should().BeTrue();
+        }
+
+        [TestMethod]
+        public void TestTryAdd()
+        {
+            var tx1 = CreateTransaction();
+            _unit.TryAdd(tx1.Hash, tx1);
```

Comment:
` _unit.TryAdd(tx1.Hash, tx1).Should().BeTrue();` here ?

Reply:
Yes, of course. Missed the check here.